### PR TITLE
Add Exponential Search In kotlin

### PR DESCRIPTION
**Pull request by hemant2705**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
I have Added Exponential Search in Kotlin.
I have added all necessary comments to make code more readable.

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#4536 
(Write your answer here.)

### CODE REVIEW

**Comment by hemant2705** [comment]
Can Anyone please review it?

File: Kotlin/search/ExponentialSearch.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  
+  System.out.println((result < 0) ? "Element is not present in array" :
+        "Element is present at index " + result);
+ }
+}
+
+//Output =  Element is present at index 3
+//Time Complexity : O(Log n)
+//Auxiliary Space :Recursive Binary Search O(Log n) space.
+//Iterative Binary Search O(1) space.
```
Comment (HarshCasper) [inline Kotlin/search/ExponentialSearch.kt]: Use Multi-Line Comments

File: Kotlin/search/ExponentialSearch.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  int i = 1;
+  while (i < n && arr[i] <= x)
+   i = i*2;
+ 
+  return Arrays.binarySearch(arr, i/2, Math.min(i, n), x);
+ }
+ 
+ public static void main(String args[])
+ {
+  int arr[] = {2, 3, 4, 10, 40};
```
Comment (HarshCasper) [inline Kotlin/search/ExponentialSearch.kt]: Use dynamic input

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by hemant2705** [comment]
Changes committed !
Can you please  go through it.

**Review by ankitaggarwal23 — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**